src/components/connect/operation/send/SendTxHintBlock.js

```diff
@@ -195,13 +198,27 @@ export default class SendTxHintBlock extends React.Component<Props> {
       const firstStep = request.getAndIncrementStep();
       const secondStep = request.getAndIncrementStep();
       const thirdStep = request.getAndIncrementStep();
+      let poolId;
+      if (request.deviceVersion === undefined) {
+        throw new Error('unexpect null deviceVersion');
+      }
```

Comment:
Are we sure by the time this code is executing that the version of the device will definitely be queried and available? 🤔 Am I getting it right from the `ConnectStore` that the version is being read-out automatically at the very first step when connecting with the device?

Reply:
Yes. The reasoning is that `ConnectStore._detectLedgerDevice` sets  `deviceVersion` before changes the progress state to `DEVICE_FOUND`(https://github.com/Emurgo/yoroi-extension-ledger-connect-vnext/blob/yushi/bech32-pool-id/src/stores/ConnectStore.js#L159), but `components/connect/ConnectBlock.js` renders `OperationBlock` (which renders `SendTxHintBlock`) only when progress state is `DEVICE_FOUND`. (https://github.com/Emurgo/yoroi-extension-ledger-connect-vnext/blob/yushi/bech32-pool-id/src/components/connect/ConnectBlock.js#L100) So it's safe to assume that `deviceVersion` is set.

Reply:
Perfect! Thank you, @yushih!